proto/profile/network.proto

```diff
@@ -66,53 +92,15 @@ message NetworkConfigChange {
   // token provisioned via the controller.
   string server_token = 1;
 
-  // New or updated network configuration for one or more network ports.
-  // Ports not listed here will keep their current configuration unchanged.
-  NetworkConfig config_changes = 2;
-}
-
-// ConfigSource specifies where the network configuration originated.
-enum ConfigSource {
-  // The origin of the configuration is unknown.
-  CONFIG_SOURCE_UNSPECIFIED = 0;
-  // Received device configuration directly from the controller over HTTPS.
-  CONFIG_SOURCE_CONTROLLER = 1;
-  // Exported device config from the controller with the controller's signature,
-  // and installed via the Single-Use EVE installer as the device's initial configuration.
-  CONFIG_SOURCE_BOOTSTRAP = 2;
-  // Manually provided network-only config in JSON format and embedded into the EVE installer.
-  CONFIG_SOURCE_OVERRIDE = 3;
-  // Exported device config from the controller with the controller's signature,
-  // imported into the Local Operator Console (LOC), and fetched by the device over HTTPS.
-  CONFIG_SOURCE_LOC = 4;
-  // Created using a Local Profile Server (LPS) application and fetched
-  // by the device using the LPS API (currently HTTP-only).
-  CONFIG_SOURCE_LPS = 5;
-}
+  // Network configuration changes to be applied to one or more ports.
+  // Ports listed here have new or updated configuration; other ports may
+  // remain unchanged or be listed in `reverted_ports` for rollback.
+  repeated NetworkPortConfig changed_ports = 2;
 
-// NetworkConfig represents a collection of network port configurations.
-// When included in a NetworkConfigChange message, it should contain only
-// the ports that require updated configurations.
-message NetworkConfig {
-  // List of network port configurations.
-  repeated NetworkPortConfig ports = 1;
-
-  // Identifies the origin of this configuration.
-  ConfigSource config_source = 2;
-
-  // Timestamp indicating when this configuration was applied.
-  // When included in a NetworkConfigChange message, this field should be left empty.
-  // EVE will automatically populate it with the time the configuration was received
-  // from LPS.
-  google.protobuf.Timestamp submitted_at = 3;
-
-  // Error message describing why the last submitted configuration was
-  // not fully/partially applied.
-  // This may include validation errors for configurations received from either
-  // the controller or LPS, or permission-related errors for configuration changes
-  // requested by LPS. Empty if the last submitted configuration was applied successfully.
-  // For port-specific validation/permission errors, see `NetworkPortConfig.error_message`
-  string error_message = 4;
+  // Ports for which locally applied configuration should be removed,
+  // reverting to the controller-provided configuration.
+  // Identified by port logical labels.
```

Comment:
This doesn't sound like a very declarative way to express things. 
Raises question like for how long something should be included in reverted_ports. Forever if the LPS has ever made an override for that port? Until it sees something coming back for EVE?

An alternative approach would be to have LPS have to include all the ports (which it is allowed to change - presumable the LPS knows that from some message) in NetworkConfigChange and for the ones which it has no overrride it sets a boolean override = false.